src/eduid_idp/cache.py

```diff
@@ -232,16 +232,16 @@ class ExpiringCacheCommonSession(ExpiringCache):
     def __init__(self, name: str, logger: logging.Logger, ttl: int, config: IdPConfig, secret: str):
         super(ExpiringCacheCommonSession, self).__init__(name, logger, ttl, lock=None)
 
-        redis_cfg = {'REDIS_PORT': config.redis_port,
-                     'REDIS_DB': config.redis_db,
+        redis_cfg = {'REDIS_PORT': config['REDIS_PORT'],
+                     'REDIS_DB': config['REDIS_DB'],
                      }
-        if config.redis_sentinel_hosts:
-            redis_cfg.update({'REDIS_SENTINEL_HOSTS': config.redis_sentinel_hosts,
-                              'REDIS_SENTINEL_SERVICE_NAME': config.redis_sentinel_service_name,
+        if config.get('REDIS_SENTINEL_HOSTS'):
```

Comment:
I think this should be if 'REDIS_SENTINEL_HOSTS' in config:

Reply:
In this cases, where the type of the setting is Optional[whatever] and the default is None, I am keeping the "config.get(...)" rather than the "... in config", otherwise we would enter the conditional when the setting is not configured and defaults to None